src/main/java/org/mockito/internal/invocation/MatchersBinder.java

```diff
@@ -28,10 +28,14 @@ public InvocationMatcher bindMatchers(ArgumentMatcherStorage argumentMatcherStor
         for (LocalizedMatcher m : lastMatchers) {
             matchers.add(m.getMatcher());
         }
+        return createMatcher(invocation, matchers);
+    }
+
+    protected InvocationMatcher createMatcher(Invocation invocation, List<ArgumentMatcher> matchers) {
         return new InvocationMatcher(invocation, matchers);
     }
 
-    private void validateMatchers(Invocation invocation, List<LocalizedMatcher> lastMatchers) {
+    protected void validateMatchers(Invocation invocation, List<LocalizedMatcher> lastMatchers) {
```

Comment:
What is the reasoning for making this protected? It is an internal class, so it should not be subclassed.

Reply:
I currently need to extend it to replace the InvocationMatcher, so I made it protected just in case, but I'm happy to leave it private if you prefer so

Reply:
@TimvdLippe, I'm ok with changing private method to protected on an internal type so that mockito-scala integration is easier in the short term. @bbonanno is aware that we don't guarantee compatibility for internal API in the long term.